### PR TITLE
fix deploy code vm type compatibility

### PR DESCRIPTION
**Pull request by laizy**

### CODE REVIEW

File: core/payload/deploy_code.go [hunk 1]
```diff
@@ -54,6 +65,30 @@ func (dc *DeployCode) Address() common.Address {
 	return dc.address
 }
 
+func (dc *DeployCode) SetVmType(ty VmType) {
+	dc.vmFlags = byte(ty)
+}
```
Comment (carltraveler) [inline core/payload/deploy_code.go:70]: check ty here?

File: core/payload/deploy_code.go [hunk 2]
```diff
@@ -252,10 +287,13 @@ func CreateDeployCode(code []byte,
 	author []byte,
 	email []byte,
 	desc []byte) (*DeployCode, error) {
+	if vmType > 255 {
+		return nil, fmt.Errorf("wrong vm flags: %d", vmType)
+	}
```
Comment (carltraveler) [inline core/payload/deploy_code.go:292]: here is redundant. validateDeployCode already check

**Review by carltraveler — APPROVED** [review]

File: core/payload/deploy_code.go (hunk 2 above)
Comment (laizy) [reply core/payload/deploy_code.go:292]: this checks uint32 -> byte convension.

File: core/payload/deploy_code.go (hunk 1 above)
Comment (laizy) [reply core/payload/deploy_code.go:70]: ty is `VmType`, which is guranteed to be valid.

**Pull request merged**